torch/distributed/_spmd/data_parallel.py

```diff
@@ -673,36 +707,39 @@ def mark_data_parallel_shardings(
             raise RuntimeError(f"op code {node.op} not supported")
 
 
-def _to_local_shard(
-    full_tensor: torch.Tensor, spec: DTensorSpec, from_local: bool = False
-) -> torch.Tensor:
+def _partition_val(val: Any, spec: DTensorSpec) -> Any:
     """
-    util function to convert a full tensor to its local component
+    util function to convert a full tensor val to its local component
     """
-    local_shard = full_tensor
-    if from_local:
-        # flag indicate the tensor is already local, we don't need to do anything
+    if isinstance(val, torch.Tensor):
+        local_shard = val
+        if val.ndim == 0:
+            # If it's already a scalar tensor, it is already local, we don't
+            # need to do anything
+            return local_shard
+
+        for idx, placement in enumerate(spec.placements):
+            if placement.is_shard():
+                placement = cast(Shard, placement)
+                num_chunks = spec.mesh.size(dim=idx)
+                my_coord = spec.mesh.get_coordinate()
+                assert my_coord is not None, "current rank not in mesh!"
+                my_coord_on_mesh_dim = my_coord[idx]
+                local_shard = placement._split_tensor(
```

Comment:
split tensor seems return a tuple of two elements?

https://github.com/pytorch/pytorch/blob/3afa60bf0f849dd61c49b66a937bb214bcd98f38/torch/distributed/_tensor/placement_types.py#L89-L91

Reply:
Yep I used `[0]` at the end of get the `shard_list`